### PR TITLE
errors: print a more helpful message when ssl fails

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Proposed Commit Message

> errors: print a more helpful message when ssl fails
> 
> Fixes: #1618

## Test Steps

* In some way, mess with ssl's ability to verify a certificate.
    * For example, uninstall `ca-certificates`
* Try to run any `ua` command that talks to the network (attach, fix, etc.)
* Verify that a descriptive and actionable message is printed

## Desired commit type::

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist:

 - [ ] I have updated or added any unit tests accordingly
 - [x] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Comment by orndorffgrant** [comment]
I don't believe any of the jenkins test failures are related to this PR, but I will look into them.

File: features/unattached_commands.feature [hunk 1]
```diff
@@ -196,6 +196,23 @@ Feature: Command behaviour when unattached
            | groovy   | no           |
            | hirsute  | no           |
 
+
+    @series.focal
```
Comment (lucasmoura) [inline features/unattached_commands.feature:200]: Can we have that test for all releases ?

File: features/unattached_commands.feature [hunk 2]
```diff
@@ -196,6 +196,23 @@ Feature: Command behaviour when unattached
...
+    Scenario Outline: Useful SSL failure message when there aren't any ca-certs
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I run `apt remove ca-certificates -y` with sudo
+        When I verify that running `ua fix CVE-1800-123456` `as non-root` exits `1`
+        Then I will see the following on stderr:
+            """
+            Cannot verify certificate of authentication server
+            Try installing `ca-certificates` if it isn't already and try again.
+            Failed to access URL: https://ubuntu.com/security/cves/CVE-1800-123456.json
```
Comment (orndorffgrant) [inline features/unattached_commands.feature:209]: Here is an example of the new message we'll print if there is an SSL verification error. I just realized I used backticks out of habit and not sure if thats what we want here. Also, should we add another line along the lines of "If that doesn't work, check your openssl configuration." - What do you think? @mitechie @lucasmoura @blackboxsw

File: features/unattached_commands.feature (hunk 1 above)
Comment (orndorffgrant) [reply features/unattached_commands.feature:200]: Yep good catch! I had just put one for getting it working, but will change this to "@series.all"

File: features/unattached_commands.feature (hunk 2 above)
Comment (lucasmoura) [reply features/unattached_commands.feature:209]: I think dropping the backticks and adding the extra line is fine.
But let's hear from @mitechie and @blackboxsw on this as well

Comment (mitechie) [reply features/unattached_commands.feature:209]: A few comments. I agree that backticks should be quotes instead. 

I'd prefer if the error came first "Failed to access URL:..." and then the other material is the details about the said error. 

Can we tell if the ca-certificates package is installed or not vs leaving it hanging with ambiguity?

Comment (orndorffgrant) [reply features/unattached_commands.feature:209]: Ack: will change to quotes and will put "Failed to access..." first.

We could tell if ca-certificates is installed or not - I just wasn't sure how far down the path of debugging the problem we think uaclient should go (e.g. if ca-certificates is installed, do we check if the openssl config file is valid? do we check to see if the cert we got from contracts.canonical.com is what we expected?).

I think checking if ca-certificates is installed or not is a reasonable step for uaclient to take, and we can print a different message accordingly ("Please install ca-certificates" vs "Check your openssl configuration"). We probably shouldn't go any further though. What do you think @mitechie @blackboxsw @lucasmoura ?

Comment (mitechie) [reply features/unattached_commands.feature:209]: If it's easy to check the package specifically we know we hit that case for a user and so it seems like a reasonable step to get the user as close to the problem as possible. Thanks for the updates.

**Comment by lucasmoura** [comment]
@orndorffgrant That looks really good, but the new integration test is failing on CI.
After we fix that, I we can land this without an issue

**Comment by orndorffgrant** [comment]
@lucasmoura Ah thanks, taking a look now

**Comment by orndorffgrant** [comment]
@lucasmoura Okay I changed the test slightly and it worked locally. Lets let jenkins try it out

**Comment by orndorffgrant** [comment]
@lucasmoura Alright several integration test failures, but none seem related to this PR.
The xenial ones are known failures right now.
The focal vm one is a forklimit error - likely something to do with jenkins being overworked?
The bionic failures look like its running the current version of the tests against an older version of UAClient. Several fixed things are failing beyond just the things touched in this PR. Any ideas what couldve caused this?

**Comment by lucasmoura** [comment]
@orndorffgrant my hypothesis for bionic is that it is running the 27.0.2 version of uaclient already and this causes the problem #1637 is trying to solve

For the focal vm test, that's new to me. I will need to take a look on it.

However, I don't think we should block this feature because of it

**Comment by lucasmoura** [comment]
@orndorffgrant just confirmed that this is the bionic issue. If we launch a daily bionic container, we can see that uaclient has version 27.0.2

**Review by lucasmoura — APPROVED** [review]

**Comment by orndorffgrant** [comment]
Thanks for verifying that @lucasmoura! This certainly makes #1637 more urgent

**Pull request merged**